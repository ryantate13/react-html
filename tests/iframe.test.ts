import {iframe} from '../src';

describe('iframe', () => {
    it('renders with children', () => {
        expect(iframe({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(iframe({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(iframe({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(iframe()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(iframe({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(iframe({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(iframe({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(iframe({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(iframe({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(iframe({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(iframe({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(iframe({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(iframe({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(iframe({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(iframe({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(iframe({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(iframe({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(iframe({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(iframe({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(iframe({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(iframe({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(iframe({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(iframe({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(iframe({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(iframe({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(iframe({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(iframe({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(iframe({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(iframe({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(iframe({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(iframe({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(iframe({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(iframe({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(iframe({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(iframe({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(iframe({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(iframe({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(iframe({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(iframe({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(iframe({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(iframe({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(iframe({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(iframe({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(iframe({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(iframe({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(iframe({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(iframe({translate: "no"})).toMatchSnapshot();
    });

    it('renders with allow="self"', () => {
        expect(iframe({allow: "self"})).toMatchSnapshot();
    });

    it('renders with allowFullScreen=true', () => {
        expect(iframe({allowFullScreen: true})).toMatchSnapshot();
    });

    it('renders with allowFullScreen=false', () => {
        expect(iframe({allowFullScreen: false})).toMatchSnapshot();
    });

    it('renders with height=1', () => {
        expect(iframe({height: 1})).toMatchSnapshot();
    });

    it('renders with name="_blank"', () => {
        expect(iframe({name: "_blank"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-forms"', () => {
        expect(iframe({sandbox: "allow-forms"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-modals"', () => {
        expect(iframe({sandbox: "allow-modals"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-orientation-lock"', () => {
        expect(iframe({sandbox: "allow-orientation-lock"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-pointer-lock"', () => {
        expect(iframe({sandbox: "allow-pointer-lock"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-popups"', () => {
        expect(iframe({sandbox: "allow-popups"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-popups-to-escape-sandbox"', () => {
        expect(iframe({sandbox: "allow-popups-to-escape-sandbox"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-presentation"', () => {
        expect(iframe({sandbox: "allow-presentation"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-same-origin"', () => {
        expect(iframe({sandbox: "allow-same-origin"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-scripts"', () => {
        expect(iframe({sandbox: "allow-scripts"})).toMatchSnapshot();
    });

    it('renders with sandbox="allow-top-navigation"', () => {
        expect(iframe({sandbox: "allow-top-navigation"})).toMatchSnapshot();
    });

    it('renders with src="https://google.com/"', () => {
        expect(iframe({src: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with srcDoc="iframe"', () => {
        expect(iframe({srcDoc: "iframe"})).toMatchSnapshot();
    });

    it('renders with srcDoc="srcdoc"', () => {
        expect(iframe({srcDoc: "srcdoc"})).toMatchSnapshot();
    });

    it('renders with width=1', () => {
        expect(iframe({width: 1})).toMatchSnapshot();
    });
});