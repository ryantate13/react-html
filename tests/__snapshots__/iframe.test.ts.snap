// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iframe renders arrays of children 1`] = `
<iframe>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</iframe>
`;

exports[`iframe renders with accessKey="a,b,c,d" 1`] = `
<iframe
  accessKey="a,b,c,d"
/>
`;

exports[`iframe renders with allow="self" 1`] = `
<iframe
  allow="self"
/>
`;

exports[`iframe renders with allowFullScreen=false 1`] = `
<iframe
  allowFullScreen={false}
/>
`;

exports[`iframe renders with allowFullScreen=true 1`] = `
<iframe
  allowFullScreen={true}
/>
`;

exports[`iframe renders with autoCapitalize="characters" 1`] = `
<iframe
  autoCapitalize="characters"
/>
`;

exports[`iframe renders with autoCapitalize="none" 1`] = `
<iframe
  autoCapitalize="none"
/>
`;

exports[`iframe renders with autoCapitalize="off" 1`] = `
<iframe
  autoCapitalize="off"
/>
`;

exports[`iframe renders with autoCapitalize="on" 1`] = `
<iframe
  autoCapitalize="on"
/>
`;

exports[`iframe renders with autoCapitalize="sentences" 1`] = `
<iframe
  autoCapitalize="sentences"
/>
`;

exports[`iframe renders with autoCapitalize="words" 1`] = `
<iframe
  autoCapitalize="words"
/>
`;

exports[`iframe renders with children 1`] = `
<iframe>
  TEST
</iframe>
`;

exports[`iframe renders with className="foo bar baz" 1`] = `
<iframe
  className="foo bar baz"
/>
`;

exports[`iframe renders with contentEditable="false" 1`] = `
<iframe
  contentEditable="false"
/>
`;

exports[`iframe renders with contentEditable="true" 1`] = `
<iframe
  contentEditable="true"
/>
`;

exports[`iframe renders with dir="auto" 1`] = `
<iframe
  dir="auto"
/>
`;

exports[`iframe renders with dir="ltr" 1`] = `
<iframe
  dir="ltr"
/>
`;

exports[`iframe renders with dir="rtl" 1`] = `
<iframe
  dir="rtl"
/>
`;

exports[`iframe renders with draggable="false" 1`] = `
<iframe
  draggable="false"
/>
`;

exports[`iframe renders with draggable="true" 1`] = `
<iframe
  draggable="true"
/>
`;

exports[`iframe renders with empty attributes 1`] = `<iframe />`;

exports[`iframe renders with height=1 1`] = `
<iframe
  height={1}
/>
`;

exports[`iframe renders with hidden=false 1`] = `
<iframe
  hidden={false}
/>
`;

exports[`iframe renders with hidden=true 1`] = `
<iframe
  hidden={true}
/>
`;

exports[`iframe renders with id="TEST" 1`] = `
<iframe
  id="TEST"
/>
`;

exports[`iframe renders with inputMode="decimal" 1`] = `
<iframe
  inputMode="decimal"
/>
`;

exports[`iframe renders with inputMode="email" 1`] = `
<iframe
  inputMode="email"
/>
`;

exports[`iframe renders with inputMode="none" 1`] = `
<iframe
  inputMode="none"
/>
`;

exports[`iframe renders with inputMode="numeric" 1`] = `
<iframe
  inputMode="numeric"
/>
`;

exports[`iframe renders with inputMode="search" 1`] = `
<iframe
  inputMode="search"
/>
`;

exports[`iframe renders with inputMode="tel" 1`] = `
<iframe
  inputMode="tel"
/>
`;

exports[`iframe renders with inputMode="text" 1`] = `
<iframe
  inputMode="text"
/>
`;

exports[`iframe renders with inputMode="url" 1`] = `
<iframe
  inputMode="url"
/>
`;

exports[`iframe renders with is="foo" 1`] = `
<iframe
  is="foo"
/>
`;

exports[`iframe renders with itemID="https://google.com/" 1`] = `
<iframe
  itemID="https://google.com/"
/>
`;

exports[`iframe renders with itemProp="https://google.com/" 1`] = `
<iframe
  itemProp="https://google.com/"
/>
`;

exports[`iframe renders with itemRef="foo bar" 1`] = `
<iframe
  itemRef="foo bar"
/>
`;

exports[`iframe renders with itemScope=false 1`] = `
<iframe
  itemScope={false}
/>
`;

exports[`iframe renders with itemScope=true 1`] = `
<iframe
  itemScope={true}
/>
`;

exports[`iframe renders with itemType="https://google.com/" 1`] = `
<iframe
  itemType="https://google.com/"
/>
`;

exports[`iframe renders with lang="" 1`] = `
<iframe
  lang=""
/>
`;

exports[`iframe renders with lang="es-419" 1`] = `
<iframe
  lang="es-419"
/>
`;

exports[`iframe renders with name="_blank" 1`] = `
<iframe
  name="_blank"
/>
`;

exports[`iframe renders with sandbox="allow-forms" 1`] = `
<iframe
  sandbox="allow-forms"
/>
`;

exports[`iframe renders with sandbox="allow-modals" 1`] = `
<iframe
  sandbox="allow-modals"
/>
`;

exports[`iframe renders with sandbox="allow-orientation-lock" 1`] = `
<iframe
  sandbox="allow-orientation-lock"
/>
`;

exports[`iframe renders with sandbox="allow-pointer-lock" 1`] = `
<iframe
  sandbox="allow-pointer-lock"
/>
`;

exports[`iframe renders with sandbox="allow-popups" 1`] = `
<iframe
  sandbox="allow-popups"
/>
`;

exports[`iframe renders with sandbox="allow-popups-to-escape-sandbox" 1`] = `
<iframe
  sandbox="allow-popups-to-escape-sandbox"
/>
`;

exports[`iframe renders with sandbox="allow-presentation" 1`] = `
<iframe
  sandbox="allow-presentation"
/>
`;

exports[`iframe renders with sandbox="allow-same-origin" 1`] = `
<iframe
  sandbox="allow-same-origin"
/>
`;

exports[`iframe renders with sandbox="allow-scripts" 1`] = `
<iframe
  sandbox="allow-scripts"
/>
`;

exports[`iframe renders with sandbox="allow-top-navigation" 1`] = `
<iframe
  sandbox="allow-top-navigation"
/>
`;

exports[`iframe renders with slot="TEST" 1`] = `
<iframe
  slot="TEST"
/>
`;

exports[`iframe renders with spellCheck="false" 1`] = `
<iframe
  spellCheck="false"
/>
`;

exports[`iframe renders with spellCheck="true" 1`] = `
<iframe
  spellCheck="true"
/>
`;

exports[`iframe renders with src="https://google.com/" 1`] = `
<iframe
  src="https://google.com/"
/>
`;

exports[`iframe renders with srcDoc="iframe" 1`] = `
<iframe
  srcDoc="iframe"
/>
`;

exports[`iframe renders with srcDoc="srcdoc" 1`] = `
<iframe
  srcDoc="srcdoc"
/>
`;

exports[`iframe renders with style={"color":"#000000"} 1`] = `
<iframe
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`iframe renders with tabIndex=1 1`] = `
<iframe
  tabIndex={1}
/>
`;

exports[`iframe renders with title="TEST" 1`] = `
<iframe
  title="TEST"
/>
`;

exports[`iframe renders with translate="no" 1`] = `
<iframe
  translate="no"
/>
`;

exports[`iframe renders with translate="yes" 1`] = `
<iframe
  translate="yes"
/>
`;

exports[`iframe renders with width=1 1`] = `
<iframe
  width={1}
/>
`;

exports[`iframe renders without attributes 1`] = `<iframe />`;
